File: desktop-client/src/main/java/uni/cc4p1/client/model/Message.java
package uni.cc4p1.client.model;

public record Message(int payloadLen, MessageType type, short senderId, short receiverId) {

    @Override
    public String toString() {
        return String.format("Header [Payload: %d bytes, Tipo: %s, Emisor: %d, Receptor: %d]",
                payloadLen, type, senderId, receiverId);
    }
}

File: mobile-client/app/src/main/java/uni/cc4p1/client/model/Message.kt
package uni.cc4p1.client.model

data class Message(
    val payloadLen: Int,
    val type: MessageType,
    val senderId: Short,
    val receiverId: Short
) {
    override fun toString(): String =
        "Header [Payload: ${payloadLen}B, Type: $type, Sender: $senderId, Receiver: $receiverId]"
}
